internal/hcp/internal/controllers/link/controller.go

```diff
@@ -184,13 +163,7 @@ func (r *linkReconciler) Reconcile(ctx context.Context, rt controller.Runtime, r
 	if accessLevel != pbhcp.AccessLevel_ACCESS_LEVEL_GLOBAL_READ_ONLY {
 		_, err = bootstrap.LoadManagementToken(ctx, rt.Logger, hcpClient, r.dataDir)
 		if err != nil {
-			rt.Logger.Error("error loading management token", "error", err)
-			newStatus := &pbresource.Status{
-				ObservedGeneration: res.Generation,
-				Conditions:         []*pbresource.Condition{ConditionFailed},
-			}
-
-			r.writeStatusIfNotEqual(ctx, rt, res, newStatus)
+			linkingFailed(ctx, rt, res, err)
```

Comment:
should we catch and log these (and similar) errors above? Just in case writing the status fails for some reason, at least there'd be some indication of that.